src/main/java/com/thoughtworks/gocd/secretmanager/vault/builders/VaultConfigBuilder.java

```diff
@@ -26,12 +26,14 @@
 
 public class VaultConfigBuilder {
     public VaultConfig configFrom(SecretConfig secretConfig) throws VaultException {
-        return new VaultConfig()
+        VaultConfig request = new VaultConfig()
                 .address(secretConfig.getVaultUrl())
                 .openTimeout(secretConfig.getConnectionTimeout())
                 .readTimeout(secretConfig.getReadTimeout())
-                .sslConfig(sslConfig(secretConfig).build())
-                .build();
+                .sslConfig(sslConfig(secretConfig).build());
+        if (isNotBlank(secretConfig.getNameSpace()))
```

Comment:
Can you please add a test for these changes.

Reply:
Is there something specific you recommend testing? Happy to throw in a bit more to the tests, but not really sure what we're looking for.

Reply:
Is there something specific you recommend testing? Happy to throw in a bit more to the tests, but not really sure what we're looking for.

Reply:
I would verify that the Builder would return a VaultConfig with namespace if provided, something similar to - https://github.com/gocd/gocd-vault-secret-plugin/blob/master/src/test/java/com/thoughtworks/gocd/secretmanager/vault/builders/AppRoleVaultConfigBuilderTest.java#L34